tests/cl-go-client/main.go

```diff
@@ -231,6 +238,78 @@ func swapGivenLargeAmountsBothDirections(igniteClient cosmosclient.Client, poolI
 	log.Println("finished swapping, num swaps done", numSwaps)
 }
 
+func createExternalCLIncentive(igniteClient cosmosclient.Client, poolId uint64, gaugeCoins sdk.Coins, expectedEpochIdentifier string) {
+	var (
+		randAccountNum = rand.Intn(8) + 1
+		accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
+	)
+
+	epochsQueryClient := epochstypes.NewQueryClient(igniteClient.Context())
+	currentEpochResponse, err := epochsQueryClient.CurrentEpoch(context.Background(), &epochstypes.QueryCurrentEpochRequest{
+		expectedEpochIdentifier,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("current epoch", currentEpochResponse.CurrentEpoch, "epoch identifier", expectedEpochIdentifier)
+
+	log.Println("querying epoch info. Note that incentives are distributed at the end of epoch. That's why it matters.")
+	epochInfosRespose, err := epochsQueryClient.EpochInfos(context.Background(), &epochstypes.QueryEpochsInfoRequest{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(epochInfosRespose.Epochs) > 0 {
+		lastEpochInfo := epochInfosRespose.Epochs[len(epochInfosRespose.Epochs)-1]
+		log.Println("epoch duration", lastEpochInfo, "next epoch start time", lastEpochInfo.StartTime.Add(lastEpochInfo.Duration))
+	} else {
+		log.Println("could not find information about previous epoch. If duration is too long, this test might be infeasible")
```

Comment:
what do you mean by "if the duration is too long"? Are you referring to the epoch duration? This is the case where there aren't any epoch registered though right?

Reply:
Like if it is 1 full day.

The script is currently set up for localosmosis where epoch duration is 60 seconds.